### PR TITLE
Add limitation to reimbursement hook

### LINKED ISSUE #27: Remove reimbursement_success_hooks and reimbursement_failed_hooks
Detect reimbursement_success_hooks and reimbursement_failed_hooks

https://github.com/nebulab/rubocop-solidus/pull/9

solidus minimum version: 2.11

Deprecation Pr: https://github.com/solidusio/solidus/pull/3541

### PR DESCRIPTION
**Pull request by MassimilianoLattanzio**
Add minimum Solidus version limitation to the reimbursement hook because the deprecation is introduced starting from Solidus 2.11.

Closes #27.

### CODE REVIEW

**Review by abhishekgupta5 — APPROVED** [review]
LFTM

**Review by piyushswain — APPROVED** [review]

File: spec/rubocop/cop/solidus/reimbursement_hook_deprecated_spec.rb [hunk 1]
```diff
@@ -30,4 +30,20 @@
...
+        reimbursement_success_hooks.each
+      RUBY
+    end
+
+    it 'avoid to register an offense when using `#reimbursement_failed_hooks.each`' do
+      expect_no_offenses(<<~RUBY)
+        reimbursement_failed_hooks.each
+      RUBY
+    end
+  end
```
Comment (DanielePalombo) [inline spec/rubocop/cop/solidus/reimbursement_hook_deprecated_spec.rb:48]: It looks to me that these tests are not related with the change.
I would suggest to remove them, `minimum_solidus_version 2.11` it's just a configuration and the behavior should be tested on [add_offense](https://github.com/nebulab/rubocop-solidus/blob/a34f822ad28d6b56ea962146de10905e40749d9c/lib/rubocop/cop/mixin/target_solidus_version.rb#L63-L67) method.

**Review by DanielePalombo — CHANGES_REQUESTED** [review]

File: spec/rubocop/cop/solidus/reimbursement_hook_deprecated_spec.rb (hunk 1 above)
Comment (MassimilianoLattanzio) [reply spec/rubocop/cop/solidus/reimbursement_hook_deprecated_spec.rb]: Tests removed after merging #42

**Comment by MassimilianoLattanzio** [comment]
Closed in favor of #43
